libfoundation/src/system-library-static.hpp

```diff
@@ -0,0 +1,244 @@
+        // registered.
+        if (p_info.__next != nullptr)
+        {
+            return;
+        }
+        
+        p_info.__next = s_chain;
+        s_chain = &p_info;
+    }
+    
+protected:
+    static MCSLibraryStaticInfo *s_chain;
+    const MCSLibraryStaticInfo *m_handle;
+};
+
+MCSLibraryStaticInfo *__MCSLibraryHandleStatic::s_chain = nullptr;
+#endif
+
+/* ================================================================
+ * Static Handle Class -- old style
+ * ================================================================ */
+
+#include <mach-o/getsect.h>
+#include <mach-o/dyld.h>
+
+class __MCSLibraryHandleStatic
+{
+    public:
+    __MCSLibraryHandleStatic(void)
+        : m_handle(nullptr)
+    {
+    }
+    
+    ~__MCSLibraryHandleStatic(void)
+    {
+    }
+    
+    bool IsDefined(void) const
+    {
+        return m_handle != nullptr;
+    }
+    
+    bool IsEqualTo(const __MCSLibraryHandleStatic& p_other) const
+    {
+        return m_handle == p_other.m_handle;
+    }
+    
+    bool Hash(void) const
+    {
+        return MCHashPointer(m_handle);
+    }
+    
+    bool CreateWithNativePath(MCStringRef p_native_path)
+    {
+        // First process the native path to remove the engine folder prefix
+        // and extension - all static libs are compiled in with just the
+        // leaf name of the library.
+        uindex_t t_last_separator = 0;
+        if (MCStringLastIndexOfChar(p_native_path,
+                                     '/',
+                                     UINDEX_MAX,
+                                     kMCStringOptionCompareExact,
+                                     t_last_separator))
+        {
+            t_last_separator += 1;
+        }
+        
+        uindex_t t_first_extension = 0;
+        if (!MCStringFirstIndexOfChar(p_native_path,
+                                      '.',
+                                      t_last_separator,
+                                      kMCStringOptionCompareExact,
+                                      t_first_extension))
+        {
+            t_first_extension = UINDEX_MAX;
+        }
+        
+        MCAutoStringRef t_leaf_name;
+        if (!MCStringCopySubstring(p_native_path,
+                                   MCRangeMake(t_last_separator,
+                                               t_first_extension - t_last_separator),
+                                   &t_leaf_name))
+        {
+            return false;
+        }
+        
+        size_t t_section_size = 0;
+        char *t_section_ptr = nullptr;
+        t_section_ptr = getsectdata("__DATA",
+                                    "__libs",
+                                    &t_section_size);
+        if (t_section_ptr != nullptr)
+        {
+            t_section_ptr += (size_t)_dyld_get_image_vmaddr_slide(0);
+            
+            MCSLibraryStaticLibInfo **t_libs =
+                    reinterpret_cast<MCSLibraryStaticLibInfo **>(t_section_ptr);
```

Comment:
Perhaps add the `[[gnu::may_alias]]` attribute to both `t_libs` and `t_section_ptr`.